src/calcium/ca.jl

```diff
@@ -0,0 +1,1447 @@
+###############################################################################
+#
+#   ca.jl : Calcium field elements
+#
+###############################################################################
+
+export ca, CalciumField, infinity, unsigned_infinity, undefined, unknown,
+       const_pi, const_euler, const_i, complex_normal_form, csgn, isnumber,
+       isimaginary, isalgebraic, isspecial, isinf, isuinf, is_signed_inf,
+       isunknown, isundefined, pow, gamma, erf, erfi, erfc
+
+###############################################################################
+#
+#   Data type and parent methods
+#
+###############################################################################
+
+parent(a::ca) = a.parent
+
+parent_type(::Type{ca}) = CalciumField
+
+elem_type(::Type{CalciumField}) = ca
+
+base_ring(a::CalciumField) = Union{}   #  ?
+
+base_ring(a::ca) = Union{}   #  ?
+
+isdomain_type(::Type{ca}) = true
+
+function deepcopy_internal(a::ca, dict::IdDict)
+   C = a.parent
+   r = C()
+   ccall((:ca_set, libcalcium), Nothing,
+         (Ref{ca}, Ref{ca}, Ref{CalciumField}), r, a, C)
+   return r
+end
+
+###############################################################################
+#
+#   Hashing
+#
+###############################################################################
+
+# todo: implement nontrivial hash functions on C
```

Comment:
Does this await something on the C side?

Reply:
Yes, it's a nontrivial function to implement (well).